receiver/filelogreceiver/filelog.go

```diff
@@ -43,22 +42,45 @@ func (f ReceiverType) Type() configmodels.Type {
 
 // CreateDefaultConfig creates a config with type and version
 func (f ReceiverType) CreateDefaultConfig() configmodels.Receiver {
+	return createDefaultConfig()
+}
+func createDefaultConfig() *FileLogConfig {
 	return &FileLogConfig{
-		ReceiverSettings: configmodels.ReceiverSettings{
-			TypeVal: configmodels.Type(typeStr),
-			NameVal: typeStr,
+		BaseConfig: stanzareceiver.BaseConfig{
+			ReceiverSettings: configmodels.ReceiverSettings{
+				TypeVal: configmodels.Type(typeStr),
+				NameVal: typeStr,
+			},
+			Parsers: stanzareceiver.ParserConfigs{},
+		},
+		Input: stanzareceiver.InputConfig{
+			"type": "file_input",
 		},
 	}
 }
 
-// Decode unmarshals configuration into a log parsing pipeline
-func (f ReceiverType) Decode(cfg configmodels.Receiver) (pipeline.Config, error) {
-	logConfig := cfg.(*FileLogConfig)
-	return stanzareceiver.DecodeOperators(logConfig.Operators)
+// BaseConfig gets the base config from config, for now
+func (f ReceiverType) BaseConfig(cfg configmodels.Receiver) stanzareceiver.BaseConfig {
+	return cfg.(*FileLogConfig).BaseConfig
 }
 
 // FileLogConfig defines configuration for the filelog receiver
 type FileLogConfig struct {
-	configmodels.ReceiverSettings `mapstructure:",squash"`
-	Operators                     stanzareceiver.OperatorConfig `mapstructure:"operators"`
+	stanzareceiver.BaseConfig `mapstructure:",squash"`
+	Input                     stanzareceiver.InputConfig `mapstructure:",remain"`
```

Comment:
So `mapstructure:",remain"` ensures that everything that is not declared by stanzareceiver.BaseConfig is captured in `Input`? Am I right?

One thing that I am a bit reluctant about is that `InputConfig` is not strongly typed, it is can contain any Key/Values, which can be error prone. Why don't we use the actual fields that are possible for filelog receiver?

UPDATE: I see you wrote this in the PR description:
>The stanza operators still do not make use of mapstructure, which forces the configuration to work with generic map[string]interface{} and map[interface{}]interface{}. This is messy and makes it difficult to write good tests. I want to clean this up soon.

Do you plan to address this in future PRs because it requires changes to log-collection library? If you plan to address in a future PR it is best to keep #2268 and #2282 open for now. (I took the liberty to put "Resolves" next to them in the PR description, but I was probably too quick, please revert back as needed).

Reply:
As you noted, `everything that is not declared by stanzareceiver.BaseConfig is captured in Input`, which is not ideal, but I don't think this has any meaningful impact on how one would configure or use the receiver.

Although I had hoped to address the migration to mapstructure as part of this PR, I found the changes to run deeper than expected. 

As a result, I pivoted to this design with the intention of satisfying the user-facing requirements of the three tickets. I believe this PR gets us far enough that we can call those tickets complete and that refactoring for the sake of maintainability can proceed independently. Issue #2334 captures this work.